### PR TITLE
[4.0][com_associations] convert to prepared statement

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes

use prepared statement for SQL

### Testing Instructions
test com_associations

### Expected result
work as before

### Actual result
N/A

### CODE REVIEW

File: administrator/components/com_associations/Helper/AssociationsHelper.php [hunk 1]
```diff
@@ -258,17 +259,19 @@ public static function getAssociationHtmlList($extensionName, $typeName, $itemId
 				if (isset($items[$langCode]['catid']))
 				{
 					$db = Factory::getDbo();
+					$id = $items[$langCode]['catid'];
```
Comment (Quy) [inline administrator/components/com_associations/Helper/AssociationsHelper.php]: No need to assign to a variable. Use it directly in the bind.

File: administrator/components/com_associations/Model/AssociationsModel.php [hunk 2]
```diff
@@ -356,33 +386,38 @@ protected function getListQuery()
...
+				->bind(':extensionname', $extensionName);
 		}
 		elseif ($typeNameExploded = explode('.', $typeName))
 		{
 			if (count($typeNameExploded) > 1 && array_pop($typeNameExploded) === 'category')
 			{
 				$section = implode('.', $typeNameExploded);
-				$query->where($db->quoteName('a.extension') . ' = ' . $db->quote($extensionName . '.' . $section));
+				$query->where($db->quoteName('a.extension') . ' = :extensionsection')
+					->bind(':extensionsection', $extensionName . '.' . $section);
```
Comment (Quy) [inline administrator/components/com_associations/Model/AssociationsModel.php]: Please check if this is ok.

File: administrator/components/com_associations/Helper/AssociationsHelper.php (hunk 1 above)
Comment (alikon) [reply administrator/components/com_associations/Helper/AssociationsHelper.php]: true

File: administrator/components/com_associations/Model/AssociationsModel.php (hunk 2 above)
Comment (alikon) [reply administrator/components/com_associations/Model/AssociationsModel.php]: it is not
thx

**Comment by Quy** [comment]
Go to Components > Multilingual Associations

File: administrator/components/com_associations/Model/AssociationsModel.php [hunk 3]
```diff
@@ -231,16 +232,28 @@ protected function getListQuery()
...
+				$db->quoteName('#__associations', 'asso'),
+				$db->quoteName('asso.id') . ' = :id'
+				. ' AND ' . $db->quoteName('asso.context') . ' = :context'
+			)
+			->join('LEFT',
+				$db->quoteName('#__associations', 'asso2'),
+				$db->quoteName('asso2.key') . ' = ' . $db->quoteName('asso.key')
 			)
-			->join('LEFT', $db->quoteName('#__associations', 'asso2') . ' ON ' . $db->quoteName('asso2.key') . ' = ' . $db->quoteName('asso.key'));
+			->bind(':id', $fields['id'], ParameterType::INTEGER)
```
Comment (Quy) [inline administrator/components/com_associations/Model/AssociationsModel.php]: `$fields['id']` is the column name and not column value. It should not be bind. This is the case with the other changes.

Comment (alikon) [reply administrator/components/com_associations/Model/AssociationsModel.php]: oh nooooo!!!!
what a shit i've done :face_with_thermometer: 
i'll fix it

Comment (alikon) [reply administrator/components/com_associations/Model/AssociationsModel.php]: fixed
or better i hope so :smile:

File: administrator/components/com_associations/Model/AssociationsModel.php [hunk 4]
```diff
@@ -231,16 +232,25 @@ protected function getListQuery()
 		$query->select($db->quoteName($fields['language'], 'language'))
 			->select($db->quoteName('l.title', 'language_title'))
 			->select($db->quoteName('l.image', 'language_image'))
-			->join('LEFT', $db->quoteName('#__languages', 'l') . ' ON ' . $db->quoteName('l.lang_code') . ' = ' . $db->quoteName($fields['language']));
+			->join('LEFT',
+				$db->quoteName('#__languages', 'l'),
+				$db->quoteName('l.lang_code') . ' = ' . $fields['language']
```
Comment (Quy) [inline administrator/components/com_associations/Model/AssociationsModel.php]: Apply `$db->quoteName()` as before this PR.

Comment (alikon) [reply administrator/components/com_associations/Model/AssociationsModel.php]: damned right.... :+1:

Comment (alikon) [reply administrator/components/com_associations/Model/AssociationsModel.php]: quoted

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 1389b94b4cfbc72422e806d868fab5593bb6bc0d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27088">issues.joomla.org/tracker/joomla-cms/27088</a>.</sub>

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 4edff27d55fdaa908f4a4df474ab61d361b8bfe1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27088">issues.joomla.org/tracker/joomla-cms/27088</a>.</sub>

**Comment by SharkyKZ** [comment]
RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27088">issues.joomla.org/tracker/joomla-cms/27088</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**